packages/cloning/cloning/clients/search_client.py

```diff
@@ -0,0 +1,132 @@
+import contextlib
+import logging
+import os
+from collections.abc import Callable
+from io import StringIO
+from urllib.error import HTTPError
+
+import httpx
+from Bio import Entrez, SeqIO
+from tenacity import (
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential_jitter,
+)
+
+from ..sequence_models import (
+    API_URL,
+    DEFAULT_TIMEOUT,
+    BioSequence,
+    BioSequences,
+    SearchResult,
+    SequenceType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+async def search_plasmids(
+    query: str,
+) -> list[SearchResult]:
+    """
+    Searches from a database of plasmids and returns best matching features or plasmids.
+
+    Args:
+        query: Keyword query search. Can negate terms with a minus sign. Supports AND/OR.
+
+    Returns:
+        A list of matching plasmids or features.
+    """
+    auth_token = os.environ["MODAL_DEPLOY_TOKEN"]
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            f"{API_URL}/search",
+            headers={
+                "Authorization": f"Bearer {auth_token}",
+            },
+            params={
+                "query": query,
+            },
+            timeout=DEFAULT_TIMEOUT,
+        )
+        response.raise_for_status()
+        data = response.json()
+    return [SearchResult(**result) for result in data]
+
+
+@retry(
+    stop=stop_after_attempt(5),
+    # jitter so that concurrent retries are less likely to collide
+    wait=wait_exponential_jitter(initial=2, max=60, jitter=4),
+    retry=retry_if_exception_type(HTTPError),
+)
+def query_entrez(endpoint: Callable, entrez_read: bool = True, **kwargs):
+    with contextlib.closing(endpoint(**kwargs)) as handle:
+        return Entrez.read(handle) if entrez_read else handle.read()
+
+
+def search_genes(query: str) -> BioSequences:
+    """Searches the NCBI nucleotide database for genes using the Entrez API and returns up to 5 results.
+
+    Args:
+        query: Search query.
+
+    Returns:
+        Up to 5 matching genes as a BioSequences object.
+    """
+    # TODO: rewrite this using httpx.AsyncClient and bypass biopython.
+    Entrez.email = "hello@futurehouse.org"  # TODO: configurable?
```

Comment:
Can you:
1. Rename this to `"hello@example.org"`
2. Make it an env var default, to resolve the `TODO`